### PR TITLE
Move CultureInfoGenerator and CultureCache projects from Localization repo

### PR DESCRIPTION
**Pull request by kirthik**
`CultureInfoList` is used in both `Localization` and `DNX` repos. Hence moving to this repo.

cc @Eilon

### CODE REVIEW

File: src/Microsoft.Extensions.CultureInfoGenerator.Sources/Program.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                return CheckFor45DotVersion(releaseKey);
+            }
+        }
+
+        // Checking the version using >= will enable forward compatibility,  
+        // however you should always compile your code on newer versions of 
+        // the framework to ensure your app works the same. 
+        private static string CheckFor45DotVersion(int releaseKey)
+        {
+            if (releaseKey >= 393273)
```
Comment (Eilon) [inline src/Microsoft.Extensions.CultureInfoGenerator.Sources/Program.cs]: Can we update this to whatever .NET 4.6 RTM uses?

File: src/Microsoft.Extensions.CultureInfoGenerator.Sources/project.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿{
```
Comment (Eilon) [inline src/Microsoft.Extensions.CultureInfoGenerator.Sources/project.json]: Hmm, why is the generator a "Sources" package? Shouldn't it be a regular old console app package that can be run?

File: src/Microsoft.Extensions.Globalization.CultureInfoCache.Sources/project.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿{
```
Comment (Eilon) [inline src/Microsoft.Extensions.Globalization.CultureInfoCache.Sources/project.json]: I believe this needs a `"shared": "*.cs",` entry, no?

**Comment by Eilon** [comment]
I think the new packages need to be added to https://github.com/aspnet/Common/blob/dev/NuGetPackageVerifier.json as well. Contact @ajaybhargavb if you need help with this. I think both the new packages are "non-shipping".

**Comment by kirthik** [comment]
Once we implement https://github.com/aspnet/Localization/issues/111, this PR is not relevant.

**Pull request closed without merging**